src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -250,14 +251,20 @@ CreateReferenceTable(Oid relationId)
  */
 static void
 ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
-						  char distributionMethod, uint32 colocationId,
-						  char replicationModel)
+						  char distributionMethod, uint32 colocationId)
 {
 	Relation relation = NULL;
 	TupleDesc relationDesc = NULL;
 	char *relationName = NULL;
 	char relationKind = 0;
 	Var *distributionColumn = NULL;
+	char replicationModel = REPLICATION_MODEL_INVALID;
+
+	/* check global replication settings before continuing */
+	EnsureReplicationSettings(InvalidOid);
+
+	replicationModel = (distributionMethod == DISTRIBUTE_BY_NONE) ?
```

Comment:
Disregard unless you agree: Personally I'd use a normal if here, but it's just a minor preference.